src/app/courses/step-view-courses/courses-step-view.component.ts

```diff
@@ -177,4 +178,36 @@ export class CoursesStepViewComponent implements OnInit, OnDestroy {
     });
   }
 
+  surveyExists(): boolean {
+    const exists = this.stepDetail && this.stepDetail.survey &&
+    this.stepDetail.survey.questions && this.stepDetail.survey.questions.length > 0;
+    return exists;
+ }
+
+ examExists(): boolean {
+   const exists = this.stepDetail && this.stepDetail.exam &&
+   this.stepDetail.exam.questions && this.stepDetail.exam.questions.length > 0;
+   return exists;
+ }
+
+examAndSurveyExist(): boolean {
+  const both = this.surveyExists() && this.examExists();
+  return both;
+ }
+
+menuTriggerButtonClick(): void {
+
+  if (!this.examAndSurveyExist()) {
```

Comment:
This can be refactored.  `goToExam` only needs to be called once, with the first parameter set based on whether there is a survey or an exam/test.

Reply:
The nested `if` statements are not ideal here, as well.  You could set a string to `both`, `survey`, or `exam` and make the conditions a little easier.